scheduler/scheduler.cpp

```diff
@@ -1798,7 +1798,11 @@ static void trigger_exit(int signum)
     } else {
         // hmm, we got killed already. try better
         static const char msg[] = "forced exit.\n";
-        write(STDERR_FILENO, msg, strlen( msg ));
+        if (write(STDERR_FILENO, msg, strlen( msg )) == -1) {
+            // Useless, but avoids GCC warn_unused_result error which
+            // can't be silenced by casting to void.
+            log_errno("Error: while writing exit message.", errno);
```

Comment:
It's not useless, it may break things. The write() is there presumably because log_errno() might block because of not being async-signal safe, especially in this case when the process doesn't seem to want to exit cleanly. There's ignore_result() somewhere in services/ , see if that helps.

Reply:
Ahh, nice.  Updated.